desc/objectives/getters.py

```diff
@@ -306,11 +356,45 @@ def _get_NAE_constraints(
     return constraints
 
 
-def maybe_add_self_consistency(thing, constraints):
+def maybe_add_self_consistency(thing, constraints):  # noqa: C901
     """Add self consistency constraints if needed."""
     params = set(unique_list(flatten_list(thing.optimizable_params))[0])
 
-    # Equilibrium
+    # check which boundary mode we are in
+    if (
+        is_any_instance(constraints, FixBoundaryR)
+        and is_any_instance(constraints, FixSectionR)
+        and is_any_instance(constraints, BoundaryRSelfConsistency)
```

Comment:
what if someone has bdryR selv consistency and FixBoundaryR? techincally that is an allowed combination, though it is rarely manually passed in